discovery-provider/src/tasks/index_user_listening_history.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import time
+from typing import DefaultDict
+import sqlalchemy as sa
+from src.models.models import UserListeningHistory
+from src.tasks.celery_app import celery
+from src.models import IndexingCheckpoints, Play
+from src.utils.helpers import UPDATE_INDEXING_CHECKPOINTS_QUERY
+
+logger = logging.getLogger(__name__)
+
+USER_LISTENING_HISTORY_TABLE_NAME = "user_listening_history"
+BATCH_SIZE = 100000 # index 100k plays at most at a time
+
+def sort_listening_history(deduped_history_dict, limit = 100):
```

Comment:
maybe we can make limit more than 100 for storage purposes since it's super cheap. maybe 1000? query-size, before table pagination, which @Kyle-Shanks  is going to be looking into early 2022, we probably want to continue to limit at 100, but I don't see why we can't go beyond in the future

Reply:
discussed and we decided to stick to 1000 considering the payload for fetching a row. JSON would be around 70 kb with 1000 listens.